test/TestProjects/FirstTest-TypeSpec/FirstTest-TypeSpec.tsp

```diff
@@ -534,3 +542,13 @@ op bodyIsModelWithProjectedEnum(@body body: ModelWithProjectedEnum ): void;
 @put
 @convenientAPI(true)
 op optionalDictionary(@body body?: Record<int32>): void;
+
+model AzureLocationModel
+{
+    location: azureLocation;
+}
+@route("/azureLocation")
+@doc("test optional dictionary.")
+@put
+@convenientAPI(true)
+op azureLocationOp(@query location: azureLocation, @header regenLocation: azureLocation, @body body?: AzureLocationModel): void;
```

Comment:
we should add a case where the body has a type of azureLocation, otherwise why we bother to add such code to deal with the case?

Reply:
also a case with response type of azurelocation

Reply:
if possible, we should have a cadl ranch case

Reply:
Yes. I am writing this. And will file a PR soon. But we can merge this first.